### DISCUSSION TITLE
Multiple possible keybindings for mini.files

### DISCUSSION THREAD

**Discussion by Dronakurl** [nvim-mini/mini.nvim#604] [Ideas]
In `mini.files`, I had the 'problem', that I wanted to exit with `<ESC>` _and_ `q`. I managed to have this behaviour with the following configuration. It would be a nice feature to have this as an option inside mini.files. When I am not alone, I could try to implement it.

```lua
      require("mini.files").setup({
        mappings = { close = "<ESC>" },
      })
      vim.api.nvim_create_autocmd({ "BufEnter" }, {
        callback = function(ev)
          if vim.bo[ev.buf].filetype == "minifiles" then
            vim.api.nvim_buf_set_keymap(ev.buf, "n", "q", ":lua require('mini.files').close()<CR>", {})
          end
        end,
      })
```

**Comment by echasnovski** [comment]
Thanks for the suggestion.

This won't be a part of 'mini.files' because buffers there are also meant to be edited inside Insert or Visual mode. Hence `<Esc>` original meaning is valuable.

Creating own buffer-local mapping is a suggested approach here. The recommended approach is to follow [an example from help](https://github.com/echasnovski/mini.nvim/blob/6d1cbcaec421b3768088b4047e78ce22e5503b99/doc/mini-files.txt#L420-L427) and use dedicated 'mini.files' events. With something like this:

```lua
vim.api.nvim_create_autocmd('User', {
  pattern = 'MiniFilesBufferCreate',
  callback = function(args)
    vim.keymap.set('n', '<Esc>', function() MiniFiles.close() end, { buffer = args.data.buf_id, desc = 'Close' })
  end,
})
```
